doc/classes/vector.adoc

```diff
@@ -0,0 +1,256 @@
+If all parameters are `nil`:: Returns `vector.zero()`
+
+=== `vector.zero() -> vector`
+Returns a new `vector` where `{x = 0, y = 0, z = 0}`
+
+=== `vector.copy(v) -> vector`
+* `v`: `vector`
+
+Returns a new `vector` where `{x = v.x, y = v.y, z = v.z}`
+
+=== `vector.from_string(s, init) -> vector, string`
+* `s`: `string`
+* `init`: `number`
+
+Returns a new `vector` parsed from `s` using `string.match()`, and any appended text after the vector data.
+
+Expects `s` to be in the following format: `(x, y, z)[..]` where `x`, `y`, and `z` are all valid numbers passed to `tonumber()` and `[..]` is the appended text which is also returned.
+There are some allowances in the parsing rules for extra whitespace as padding around elements, but anything after the closing parenthesis is returned verbatum.
+
+=== `vector.to_string(v) -> string`
+* `v`: `vector`
+
+Returns a `string` representation of `v` in the format: `(v.x, v.y, v.z)`.
+
+Each component is formatted with the printf-style `%g` flag (either floating-point or scientific notation, whichever is shorter).
+
+=== `vector.equals(a, b) -> bool`
+* `a`, `b`: `vector`
+
+Returns `true` if `a` is equivalent to `b` (all components are the same).
+
+Returns `false` otherwise.
+
+=== `vector.length(v) -> number`
+* `v`: `vector`
+
+Returns the vectorial length (total traveled traveled distance from the origin to the end) of `v`.
+
+The equation for the length is: stem:[sqrt(x^2 + y^2 + z^2)].
+
+=== `vector.normalize(v) -> vector`
+* `v`: `vector`
+
+Returns a new `vector` which is the normalized form of `v` (the vectorial length is equal to 1).
+Uses `vector.length(v)` to get the vectorial length.
+
+Specifically, if the length is 0, returns a new `vector` with all components being `0`.
+Otherwise, returns `vector.divide(v, length)`.
+
+=== `vector.floor(v) -> vector`
+* `v`: `vector`
+
+Returns a new `vector` where each component of `v` has had `math.floor()` applied to it.
+
+Literally `vector.apply(v, math.floor)`.
+
+=== `vector.round(v) -> vector`
+* `v`: `vector`
+
+Returns a new `vector` where each component of `v` has had `math.round()` applied to it.
+
+Equivalent to `vector.apply(v, math.round)`
+
+=== `vector.apply(v, func) -> vector`
+* `v`: `vector`
+* `func`: `function`
+
+Returns a new `vector` where each component of `v` has had `func()` applied to it.
+
+=== `vector.distance(a, b) -> number`
+* `a`, `b`: `vector`
+
+Returns a `number` which is equal to the distance between `a` and `b`.
+
+=== `vector.direction(pos1, pos2) -> vector`
+* `post1`, `pos2`: `vector`
+
+Returns a new, normalized `vector` equal to the direction from `pos1` to `pos2`.
+
+=== `vector.angle(a, b) -> number`
+* `a`, `b`: `vector`
+
+Returns a `number` which is equal to the angle (in radians) between `a` and `b`.
+
+Formula used is stem:[tan^-1(|bar a xx bar b|, bar a * bar b)].
+
+=== `vector.dot(a, b) -> number`
+* `a`, `b`: `vector`
+
+Returns a `number` equal to the dot product of `a` and `b`.
```

Comment:
Should we elaborate on the dot product here?

Reply:
Same for the cross product.
Noting some mathematical properties about these operations might suit very well into some sort of geometry tutorial style document, that might also mention matrices and quaternions.

Reply:
Somebody more familiar with that should probably write that sort of document, though I agree that it'd be a great thing to have here.